src/sentry/http.py

```diff
@@ -331,16 +334,18 @@ def fetch_file(url, headers=None, domain_lock_enabled=True, outfile=None):
             # TODO(dcramer): we want to be less aggressive on disabling domains
             if domain_lock_enabled:
                 cache.set(domain_key, error or '', 300)
-            logger.warning('source.disabled', extra=error)
+                logger.warning('source.disabled', extra=error)
             raise CannotFetch(error)
 
-        if outfile is not None:
-            body = None
-        else:
-            body = b''.join(contents)
         headers = {k.lower(): v for k, v in response.headers.items()}
         encoding = response.encoding
 
+        if isinstance(outfile, file):
```

Comment:
I would do it differently. If the code above creates a `BytesIO` then we set a flag to true we check down here again. I would not want the code to randomly call` getvalue()` just because I happen to pass a buffered IO object in anyways.